nxc/modules/change-password.py

```diff
@@ -0,0 +1,166 @@
+import sys
+from impacket.dcerpc.v5 import samr, epm, transport
+from impacket.dcerpc.v5.rpcrt import DCERPCException
+
+class NXCModule:
+    """
+    Module for changing or resetting user passwords
+    Module by Fagan Afandiyev and termanix
+    """
+
+    name = "change-password"
+    description = "Change or reset user passwords via various protocols"
+    supported_protocols = ["smb"]
+    opsec_safe = True
+    multiple_hosts = False
+
+    def options(self, context, module_options):
+        """
+        Required (one of):
+        NEWPASS     The new password of the user.
+        NEWNTHASH   The new NT hash of the user.
+
+        Optional:
+        OLDPASS     The old password of the user (if not specified, the current password will be used)
+        OLDNTHASH   The old NT hash of the user (if not specified, the current password will be used)
+        USER        The user account if the target is not the current user.
+
+        Examples
+        --------
+        If STATUS_PASSWORD_MUST_CHANGE or STATUS_PASSWORD_EXPIRED (Change password for current user)
+            netexec smb <DC_IP> -u username -p oldpass -M change-password -o OLDPASS='oldpass' NEWPASS='newpass'
+            netexec smb <DC_IP> -u username -H oldnthash -M change-password -o OLDNTHASH='oldnthash' NEWPASS='newpass'
```

Comment:
`OLDPASS`/`OLDNTHASH` can probably be used from connection.nthash/connection.password

Reply:
I was not aware of it being available in connection.nthash. I will soon push an update using this after I test it

Reply:
```suggestion
        # If OLDPASS or OLDHASH are not specified, default to the credentials used for authentication.
        if not self.oldpass:
            self.oldpass = connection.password
        if not self.oldhash:
            self.oldhash = connection.nthash
```

Those values are set using connection.nthash if not provided but it is required to provide a different user's old password to reset the password. That option is left for that reason

Reply:
Makes sense!

Reply:
I could remove the options in the help menu but leave as it is right now for any weird specific cases that might occur. Your point is valid though